src/core/thread/mesh_forwarder.cpp

```diff
@@ -251,18 +253,31 @@ void MeshForwarder::ScheduleTransmissionTask()
             continue;
         }
 
-        mSendMessage = child.mIndirectSendMessage;
+        mSendMessage = child.mIndirectSendInfo.mMessage;
 
         if (mSendMessage == NULL)
         {
             mSendMessage = GetIndirectTransmission(child);
-            child.mIndirectSendMessage = mSendMessage;
-            child.mFragmentOffset = 0;
+
+            child.mIndirectSendInfo.mMessage = mSendMessage;
+            child.mIndirectSendInfo.mFragmentOffset = 0;
+            child.mIndirectSendInfo.mTxAttemptCounter = 0;
```

Comment:
Thinking more about this..may not be a bad idea.. Will move this in the `GetIndirectTx()` and the rest of stuff in `PrepareIndirectTx()`...